sympy/physics/mechanics/system.py

```diff
@@ -440,12 +440,17 @@ def _parse_coordinates(self, new_coords, independent, old_coords_ind,
                 Matrix(1, len(coords_dep), coords_dep).T)
 
     @staticmethod
-    def _parse_expressions(new_expressions, old_expressions, name):
+    def _parse_expressions(new_expressions, old_expressions, name,
+                           check_negatives=False):
         """Helper to parse expressions like constraints."""
         old_expressions = old_expressions[:]
         new_expressions = list(new_expressions)  # Converts a possible tuple
-        System._check_objects(new_expressions, old_expressions, Basic,
-                              name, 'expressions')
+        if check_negatives:
+            check_exprs = old_expressions + [-expr for expr in old_expressions]
+        else:
+            check_exprs = old_expressions
```

Comment:
```suggestion
        check_exprs = old_expressions
        if check_negatives:
            check_exprs += [-expr for expr in old_expressions]
```

Reply:
Can do this, but then we should use `check_exprs = old_expressions.copy()`, otherwise you'll still be editing the original.

Reply:
Silly me. Probably fine to leave as is.